math/prime/prime_test.go

```diff
@@ -0,0 +1,95 @@
+package prime
+
+import (
+	"fmt"
+	"testing"
+)
+
+var primeList = []int{2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127}
+var testLimit = 127
+
+type primalityTest func(int64) bool
+
+func primalityTestTestingHelper(t *testing.T, name string, f primalityTest) {
+	arrayIndex := 0
+	for i := 1; i <= testLimit; i++ {
+		isPrime := i == primeList[arrayIndex]
```

Comment:
The non primes still might be an issue in the probabilistic test (unlikely but possible) - so it might be best to keep the prime vs non-prime numbers separate... Maybe write a separate test for the non-primes only for the deterministic version and the one that is being shared should only be testing the algorithm with prime numbers.

Reply:
We are currently using 40 rounds of Miller-Rabin tests. The max composite which we are testing is 125 (126 is even so will be discarded by the `isTrivial` function. A maximum of ¼ numbers below the number being tested can be liars. 125/4 = 31.25. So I think we are safe. The probability of that occurring is really low, and even if that happens, a re-run of the tests must fix it.